CONTRIBUTING.rst

```diff
@@ -0,0 +1,268 @@
+   ``metashare/env``)::
+
+    BUCKETEER_AWS_ACCESS_KEY_ID=...
+    BUCKETEER_AWS_SECRET_ACCESS_KEY=...
+    BUCKETEER_BUCKET_NAME=...
+
+3. Run ``docker-compose build`` to build/re-build all the container images.
+
+4. Run ``docker-compose up`` to start the server(s).
+
+5. Visit `<http://localhost:8080/>`_ in your browser.
+
+6. When you're done working on MetaShare, ``Ctrl-C`` in the terminal where the
+   containers are running to exit. You can also ``docker-compose down`` to clean
+   up all running containers. (``docker-compose ps`` will tell you what
+   containers are currently running.)
+
+.. _Docker Desktop (Community Edition): https://www.docker.com/products/docker-desktop
+
+Docker development tasks
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To run any development tasks (such as changing Python or JS dependencies, or
+generating or running migrations, or running a Django shell), you will need to
+run them inside the appropriate Docker image. This takes the general form
+``docker-compose run --no-deps [web or client] [command]``. In some cases, such
+as for migrations or a Django shell, you will want to omit the ``--no-deps``
+flag.
+
+You shouldn't need to run any of the setup tasks; the Docker images will take
+care of setting up a database and installing Python and JS dependencies for you.
+
+When you change Python or JS dependencies, you will need to rebuild the Docker
+images, as we store dependencies in the images for speed. This requires running
+``docker-compose build [web or client]``.
+
+Tests and linting can be run similarly::
+
+    $ docker-compose run --no-deps client yarn lintjs
+    $ docker-compose run --no-deps client yarn stylelint
+    $ docker-compose run --no-deps client yarn test
+
+    $ docker-compose run --no-deps web black manage.py metashare/ config/
+    $ docker-compose run --no-deps web isort -rc manage.py metashare/ config/
+    $ docker-compose run --no-deps web flake8 manage.py metashare/ config/
+    $ docker-compose run web pytest
+
+Logging in with Salesforce
+--------------------------
+
+To setup the Salesforce OAuth integration, run the ``populate_social_apps``
+management command. The values to use in place of the ``XXX`` and ``YYY`` flags
+can be found on the Connected App you've made in your Salesforce configuration,
+or if you're an OddBird, you can find these values in the shared Keybase team
+folder (``metashare/prod.db``)::
+
+    $ docker-compose run web python manage.py populate_social_apps --prod-id XXX --prod-secret YYY
+
+(If you are using local development, omit the ``docker-compose run web``.)
+
+You can also run it with ``--test-id`` and ``--test-secret``, or ``--cust-id``
+and ``--cust-secret``, or all three sets at once, to populate all three
+providers.
+
+Once you've logged in, you probably want to make your user a superuser. You can
+do that easily via the ``promote_superuser`` management command::
+
+    $ docker-compose run web python manage.py promote_superuser <your email>
```

Comment:
@wlonk Accessing the Django admin seems to be broken in Docker-dev:

```
Environment:


Request Method: GET
Request URL: http://localhost:8080/admin

Django Version: 2.2.2
Python Version: 3.7.3
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.sites',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'channels',
 'whitenoise.runserver_nostatic',
 'django.contrib.staticfiles',
 'django_rq',
 'scheduler',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'rest_framework',
 'rest_framework.authtoken',
 'parler',
 'metashare',
 'metashare.multisalesforce',
 'metashare.api',
 'metashare.adminapi.apps.AdminapiConfig',
 'django_js_reverse',
 'django_extensions']
Installed Middleware:
['metashare.logging_middleware.LoggingMiddleware',
 'sfdo_template_helpers.admin.middleware.AdminRestrictMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback:

File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/src/sfdo-template-helpers/sfdo_template_helpers/admin/middleware.py" in __call__
  27.                 self._validate_ip(request)

File "/src/sfdo-template-helpers/sfdo_template_helpers/admin/middleware.py" in _validate_ip
  36.             raise SuspiciousOperation(f"Disallowed IP address: {ip_addr}")

Exception Type: SuspiciousOperation at /admin
Exception Value: Disallowed IP address: 172.31.0.2
```

Reply:
Hah! So, this doesn't happen in local dev because we're proxying from `localhost` to `localhost`. Here, the two Docker containers have their own IPs. I'll see about fixing it, but in the meantime, you can access it from `http://localhost:8000/admin/`.

Reply:
Same error at :8000.

Reply:
Oh, yeah. This is because of the SFDO "if it's not localhost, make sure it's on our subnet" middleware. :-/ I'm working on a fix.

Reply:
@wlonk you can set the default allowed subnets to be inclusive. we'll set it appropriately in production.